bitbucket/service.py

```diff
@@ -20,42 +23,33 @@ def __post_init__(self):
         Raises:
             bitbucket.models.KatkaProject.DoesNotExist: in case the given id for the katka project does not exist
         """
-        self._katka_project = KatkaProject.objects.get(project_id=self.katka_project_id)
-
-        if not self._katka_project.oauth_access_token:
-            log.debug(f'No access token found for katka project {self.katka_project_id}')
-
+        self.base_url = self.base_url or settings.DEFAULT_BITBUCKET_SERVICE_LOCATION
+        self.base_url = self.base_url.rstrip('/')
         self.base_path = 'rest/api/1.0'
-        self.path = ''
-
-    @property
-    def client(self) -> requests.sessions.Session:
-        session = requests.Session()
-        session.headers.update({
-            'Authorization': f'Bearer {self._katka_project.oauth_access_token}',
-            'Accept': 'application/json',
-            'User-Agent': 'katka'
-        })
-        return session
-
-    def get(self, params: dict = None) -> dict:
-        '''
+        self.bearer_token = self.credentials_provider.access_token
```

Comment:
Should there not be a check on credentials provider existing here?

Reply:
Agree